source/Octopus.Tentacle.CommonTestUtils/OctopusPackageDownloader.cs

```diff
@@ -100,6 +102,18 @@ static async Task AttemptToDownloadPackage(string downloadUrl, string filePath,
             ValidateDownload(filePath, expectedHash);
         }
 
+        static async Task<int> ReadToBufferWithTimeout(CancellationToken cancellationToken, TimeSpan singleReadTimeout, Stream contentStream, byte[] buffer)
+        {
+            using var readCts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+            readCts.CancelAfter(singleReadTimeout); 
+            var readTask = contentStream.ReadAsync(buffer, 0, buffer.Length, readCts.Token);
+            // Don't trust that cancellation tokens will work on all dotnet versions or OSs we have to test in, so also add Task.Delay ;(
```

Comment:
Oh dear. Link?

Reply:
I don't have a link, but this is what we had to do in halibut for the reliable tentacle comms.

We found that the CT was just ignored under net48